scene/resources/convex_polygon_shape_2d.cpp

```diff
@@ -43,10 +43,38 @@ void ConvexPolygonShape2D::_update_shape() {
 	if (Geometry2D::is_polygon_clockwise(final_points)) { //needs to be counter clockwise
 		final_points.reverse();
 	}
+#ifdef DEBUG_ENABLED
+	if (!is_convex(final_points)) {
+		WARN_PRINT("Concave polygon is assigned to ConvexPolygonShape2D");
+	}
+#endif
 	PhysicsServer2D::get_singleton()->shape_set_data(get_rid(), final_points);
 	emit_changed();
 }
 
+// Check if point p3 is to the left of [p1, p2] segment or on it
+bool ConvexPolygonShape2D::left_test(const Vector2 &p1, const Vector2 &p2, const Vector2 &p3) {
+	Vector2 p12 = p2 - p1;
+	Vector2 p13 = p3 - p1;
+	// If the value of the cross product is positive or zero; p3 is to the left or on the segment, respectively
+	return p12.cross(p13) >= 0;
+}
+
+bool ConvexPolygonShape2D::is_convex(const Vector<Vector2> &p_points) {
+	// Pre-condition: Polygon is in counter-clockwise order
+	int polygon_size = p_points.size();
+	for (int i = 0; i < polygon_size && polygon_size > 3; i++) {
+		int j = (i + 1) % polygon_size;
+		int k = (j + 1) % polygon_size;
+		// If any consecutive three points fail left-test, then there is a concavity
+		if (!left_test(p_points[i], p_points[j], p_points[k])) {
+			return false;
+		}
+	}
+
+	return true;
+}
```

Comment:
These should also be in `#ifdef DEBUG_ENABLED`.

Also add dots at the end of comment sentences:
```suggestion
#ifdef DEBUG_ENABLED
// Check if point p3 is to the left of [p1, p2] segment or on it.
bool ConvexPolygonShape2D::left_test(const Vector2 &p1, const Vector2 &p2, const Vector2 &p3) {
	Vector2 p12 = p2 - p1;
	Vector2 p13 = p3 - p1;
	// If the value of the cross product is positive or zero; p3 is to the left or on the segment, respectively.
	return p12.cross(p13) >= 0;
}

bool ConvexPolygonShape2D::is_convex(const Vector<Vector2> &p_points) {
	// Pre-condition: Polygon is in counter-clockwise order.
	int polygon_size = p_points.size();
	for (int i = 0; i < polygon_size && polygon_size > 3; i++) {
		int j = (i + 1) % polygon_size;
		int k = (j + 1) % polygon_size;
		// If any consecutive three points fail left-test, then there is a concavity.
		if (!left_test(p_points[i], p_points[j], p_points[k])) {
			return false;
		}
	}

	return true;
}
#endif
```